proposal/ordered-comments.md

```diff
@@ -0,0 +1,170 @@
+meaningful comments like license headers or postprocessor directives.
+
+## Summary
+
+> This section is non-normative
+
+This proposal adjusts the order that comments are emitted relative to CSS so
+that the resulting stylesheet is in essentially the same order it would have
+been when using `@import`, without repeating stylesheets that are loaded
+multiple times.
+
+### Design Decisions
+
+#### Traversal Order
+
+There are two potential orders that could be chosen for comments. Either would
+be better than the current ordering, and either can be better than the other
+depending on the user's needs.
+
+The order we propose here is "traversal order", in which comments are emitted in
+the same order that they're evaluated. The other plausible ordering is "linked
+order", in which comments that appear directly above a `@use` or `@forward` rule
+are always emitted before the module loaded by that rule.
+
+These two orderings result in the same output when each module is loaded once,
+but once a given module is loaded multiple times with comments around it they
+become different. For example:
+
+```scss
+// _upstream.scss
+a {b: c}
+```
+
+```scss
+// styles.scss
+/* before @use */
+@use 'upstream';
+
+/* before @forward */
+@forward 'upstream';
+```
+
+produces the following outputs:
+
+```scss
+// In traversal order
+/* before @use */
+a {
+  b: c;
+}
+
+/* before @forward */
+```
+
+```scss
+// In linked order
+/* before @use */
+/* before @forward
+a {
+  b: c;
+}
+```
+
+Linked order makes sense when using comments to annotate information about
+dependencies, but it's counterproductive when a user wants to annotate the _end_
+of a module, since that comment would be considered linked to the next module
+load. Traversal order handles that case better _and_ matches the old `@import`
+behavior, so we chose to use it instead.
+
+## Procedures
+
+### Resolving a Module's Extensions
+
+Adjust the definition of [Resolving a Module's Extensions] by replacing the
+definition of the traversing procedure with the following:
+
+[Resolving a Module's Extensions]: ../spec/at-rules/extend.md#resolving-a-modules-extensions
+
+* Define a mutating recursive procedure, *traversing*, which takes a module
+  `domestic`:
+
+  * If `domestic` has already been traversed, do nothing.
+
+  * For each module `upstream` in `domestic`'s dependencies:
+
+    * For each unmarked comment in `domestic`'s CSS, if that comment originally
+      appeared before the `@use` or `@forward` rule that loaded `upstream`, add
+      a copy of that comment to `css` and then mark it.
+
+    * Traverse `upstream`.
+
+    > Because this traverses modules depth-first, it emits CSS in reverse
+    > topological order.
+
+  * Let `initial-imports` be the longest initial subsequence of top-level
+    statements in `domestic`'s CSS tree that contains only comments and
+    `@import` rules *and* that ends with an `@import` rule.
```

Comment:
Given that we take the long initial subsequence of comments and `@import` rules in `css` in the next step when looking for the insertion point, should we enforce ending with an `@import` here ?

Reply:
I think we should put this below comments, because otherwise if the entrypoint file contains no `@import`s they'll end up going before _everything_ in that file, including potentially a license comment, which wouldn't be ideal.